Compiler/NFFrontEnd/NFInst.mo

```diff
@@ -1388,5 +1419,155 @@ algorithm
+      SCode.Visibility vis2;
+      String name1,name2,env_path;
+      Option<SCode.ExternalDecl> externalDecl1,externalDecl2;
+      list<SCode.Annotation> annotationLst1,annotationLst2;
+      SCode.Comment comment1,comment2;
+      Option<SCode.Annotation> ann1,ann2;
+      list<SCode.Element> els1,els2;
+      list<SCode.Equation> nEqn1,nEqn2,inEqn1,inEqn2;
+      list<SCode.AlgorithmSection> nAlg1,nAlg2,inAlg1,inAlg2;
+      list<SCode.ConstraintSection> inCons1, inCons2;
+      list<Absyn.NamedArg> clats;
+      list<tuple<SCode.Element, DAE.Mod, Boolean>> rest;
+      tuple<SCode.Element, DAE.Mod, Boolean> first;
+      SCode.Mod mods, derivedMod;
+      DAE.Mod mod1,emod;
+      SourceInfo info1, info2;
+      Boolean b;
+      SCode.Attributes attrs;
+      Absyn.TypeSpec derivedTySpec;
+
+    // found the base class with parts
+    case (cl as SCode.CLASS(name = name2, classDef = SCode.PARTS()), classExtendsElt)
+      equation
+        name1 = inClassExtendsEltName;
+        name2 = inBaseEltName;
+
+        SCode.CLASS(_,prefixes2,encapsulatedPrefix2,partialPrefix2,restriction2,SCode.PARTS(els2,nEqn2,inEqn2,nAlg2,inAlg2,inCons2,clats,externalDecl2),comment2,info2) = cl;
+
+        SCode.CLASS(_, prefixes1, encapsulatedPrefix1, partialPrefix1, restriction1, classExtendsCdef, comment1, info1) = classExtendsElt;
+        SCode.CLASS_EXTENDS(_,mods,SCode.PARTS(els1,nEqn1,inEqn1,nAlg1,inAlg1,inCons1,_,externalDecl1)) = classExtendsCdef;
+
+        classDef = SCode.PARTS(els2,nEqn2,inEqn2,nAlg2,inAlg2,inCons2,clats,externalDecl2);
+        compelt = SCode.CLASS(name2,prefixes2,encapsulatedPrefix2,partialPrefix2,restriction2,classDef,comment2,info2);
+        vis2 = SCode.prefixesVisibility(prefixes2);
+        elt = SCode.EXTENDS(Absyn.IDENT(name2),vis2,mods,NONE(),info1);
+        classDef = SCode.PARTS(elt::els1,nEqn1,inEqn1,nAlg1,inAlg1,inCons1,clats,externalDecl1);
+        elt = SCode.CLASS(name1, prefixes1, encapsulatedPrefix1, partialPrefix1, restriction1, classDef, comment1, info1);
+      then
+        (compelt, elt);
+
+    // found the base class which is derived
+    case (cl as SCode.CLASS(name = name2, classDef = SCode.DERIVED()), classExtendsElt)
+      equation
+        name1 = inClassExtendsEltName;
+        name2 = inBaseEltName;
+
+        SCode.CLASS(_,prefixes2,encapsulatedPrefix2,partialPrefix2,restriction2,SCode.DERIVED(derivedTySpec, derivedMod, attrs),comment2,info2) = cl;
+
+        SCode.CLASS(_, prefixes1, encapsulatedPrefix1, partialPrefix1, restriction1, classExtendsCdef, comment1, info1) = classExtendsElt;
+        SCode.CLASS_EXTENDS(_,mods,SCode.PARTS(els1,nEqn1,inEqn1,nAlg1,inAlg1,inCons1,_,externalDecl1)) = classExtendsCdef;
+
+        classDef = SCode.DERIVED(derivedTySpec, derivedMod, attrs);
+        compelt = SCode.CLASS(name2,prefixes2,encapsulatedPrefix2,partialPrefix2,restriction2,classDef,comment2,info2);
+        vis2 = SCode.prefixesVisibility(prefixes2);
```

Comment:
This doesn't seem correct to me, I would assume that in a class extends the extends clause is always public.